runhouse/resources/hardware/cluster.py

```diff
@@ -1385,3 +1401,55 @@ def set_connection_defaults(self, **kwargs):
                 f"Cluster name {self.name} is a reserved name. Please use a different name which is not one of "
                 f"{RESERVED_SYSTEM_NAMES}."
             )
+
+    def share(
+        self,
+        users: Union[str, List[str]] = None,
+        access_level: Union[ResourceAccess, str] = ResourceAccess.READ,
+        visibility: Optional[Union[ResourceVisibility, str]] = None,
+        notify_users: bool = True,
+        headers: Optional[Dict] = None,
+        # Deprecated
+        access_type: Union[ResourceAccess, str] = None,
+    ) -> Tuple[Dict[str, ResourceAccess], Dict[str, ResourceAccess]]:
+
+        # save cluster and creds if not saved
+        self.save()
+
+        # share creds
```

Comment:
can we add a log here to the user saying that the underlying creds are also being shared?